### PR TITLE
Introduce integration testing

### PR DESCRIPTION
**Pull request by arbonap**
- Ignore tmp/ directory
- Add pytest-splinter and pytest-flask
- Re-wrote admin preview tests
- Test page title editing
- Specify Firefox version for Selenium compatibility

### CODE REVIEW

File: tests/demo-project/Website.lektorproject [hunk 1]
```diff
@@ -16,6 +16,6 @@ url_prefix = /de/
 [servers.production]
 enabled = yes
 name = Production
-target = rsync://myserver.com/path/to/website
+target = rsync://example.com/path/to/website
```
Comment (arbonap) [inline tests/demo-project/Website.lektorproject:19]: example.com is a reserved url for testing.

**Comment by go-bears** [comment]
@arbonap we should close this pr ?

**Comment by singingwolfboy** [comment]
@arbonap @go-bears between this pull request and #271, which one should be reviewed (and eventually merged)?

**Comment by go-bears** [comment]
@singingwolfboy  @arbonap  #271 encompasses the changes made in here.

**Comment by singingwolfboy** [comment]
That's what I thought. In that case, I'm going to close this pull request.

**Pull request closed without merging**